test/xunit.analyzers.tests/CodeAnalyzerHelper.cs

```diff
@@ -1,55 +1,33 @@
-        internal static readonly MetadataReference XunitAbstractionsReference = MetadataReference.CreateFromFile(typeof(ITest).GetTypeInfo().Assembly.Location);
-        internal static readonly MetadataReference XunitAssertReference = MetadataReference.CreateFromFile(typeof(Assert).GetTypeInfo().Assembly.Location);
-        internal static readonly MetadataReference XunitCoreReference = MetadataReference.CreateFromFile(typeof(FactAttribute).GetTypeInfo().Assembly.Location);
-        internal static readonly MetadataReference XunitExecutionReference = MetadataReference.CreateFromFile(typeof(XunitTestCase).GetTypeInfo().Assembly.Location);
+        internal static ReferenceAssemblies CurrentXunit { get; }
 
         static CodeAnalyzerHelper()
         {
-            // Xunit is a PCL linked against System.Runtime, however on the Desktop framework all types in that assembly have been forwarded to
-            // System.Core, so we need to find the assembly by name to compile without errors.
-            var referencedAssemblies = typeof(FactAttribute).Assembly.GetReferencedAssemblies();
-#if NETCOREAPP2_2
-            SystemCollectionsReference = GetAssemblyReference(referencedAssemblies, "System.Collections");
-            SystemRuntimeReference = GetAssemblyReference(referencedAssemblies, "System.Runtime");
-            SystemRuntimeExtensionsReference = GetAssemblyReference(referencedAssemblies, "System.Runtime.Extensions");
-#endif
+            var referenceAssemblies = ReferenceAssemblies.Default;
 
-            SystemThreadingTasksReference = GetAssemblyReference(referencedAssemblies, "System.Threading.Tasks");
+#if NETCOREAPP
+            // Workaround for reference assembly package not included in PackageResolverContext
+            referenceAssemblies = referenceAssemblies.AddPackages(ImmutableArray.Create(ReferenceAssemblies.Default.ReferenceAssemblyPackage));
 
-#if NETCOREAPP2_2
-            // Xunit doesn't directly reference System.Collections.NonGeneric, so we locate it relative to
-            // System.Collections.
-            SystemCollectionsNonGenericReference = MetadataReference.CreateFromFile(Path.Combine(Path.GetDirectoryName(SystemCollectionsReference.FilePath), "System.Collections.NonGeneric.dll"));
-            SystemConsoleReference = MetadataReference.CreateFromFile(Path.Combine(Path.GetDirectoryName(SystemCollectionsReference.FilePath), "System.Console.dll"));
-            SystemTextRegularExpressionsReference = MetadataReference.CreateFromFile(Path.Combine(Path.GetDirectoryName(SystemCollectionsReference.FilePath), "System.Text.RegularExpressions.dll"));
+            // Workaround for ReferenceAssemblies.NetStandard.NetStandard20 not including facade assemblies automatically
+            referenceAssemblies = referenceAssemblies.AddAssemblies(ImmutableArray.Create(
+                "System.Collections",
+                "System.Runtime",
+                "System.Threading.Tasks"));
 #endif
```

Comment:
These issues are fixed by dotnet/roslyn-sdk#403. After a new package is published with this fix, I will remove these lines and mark this pull request as ready for review.